src/app/(global)/layout.tsx

```diff
@@ -26,10 +26,16 @@ export default function GlobalLayout({
     }
   }, [setAuth]);
 
-  const articleIdMatch = pathname.match(/\/article\/edit\/(\d+)/);
-  const articleId = articleIdMatch ? articleIdMatch[1] : null;
+const isEditPage = pathname.startsWith('/article/edit/');
+const isPublishPage = pathname === '/article/publish';
+const isCreatePage = pathname === '/article/create';
```

Comment:
```suggestion
const articlePathRegex = /^\/article\/(publish(\/[a-f0-9]+)?|create|edit\/[a-f0-9]+)$/;
const shouldHideHeader =  articlePathRegex.test(pathname);
```
https://regex101.com/r/QbxKxV/1